src/components/Sidebar/Sidebar.tsx

```diff
@@ -31,6 +36,12 @@ function SidebarComponent({
 }: Props) {
   const [active, activeSet] = useStateTransmorphic<string>(state);
   const [layoutStyles, displayStyles] = useLayoutStyles(style);
+  const sidebarRef = useRef<HTMLDivElement | null>(null);
+  const [sidebarWidth, setSidebarWidth] = useState(0);
+
+  useLayoutEffect(() => {
```

Comment:
Using the layout effect to determine the panel's slide-out offset (so that it isn't obscured by the sidebar)

The sidebar width changes based on whether the items have labels or not. So instead of using static widths, this is automatic.